telepresence/proxy/deployment.py

```diff
@@ -78,9 +78,20 @@ def remove_existing_deployment():
         command.append(
             "--env=TELEPRESENCE_NAMESERVER=" + get_alternate_nameserver()
         )
-    runner.get_output(runner.kubectl(command))
-    span.end()
-    return deployment_arg, run_id
+    try:
+        runner.get_output(runner.kubectl(command))
+        span.end()
+        return deployment_arg, run_id
+    except CalledProcessError:
+        command = [
+            "get",
+            "deployment",
+            deployment_arg,
+        ]
+        deployment_already_exists = runner.get_output(runner.kubectl(command))
+        if deployment_already_exists:
+            raise runner.fail("Deployment already exixts")
+        return deployment_arg, run_id
```

Comment:
What will this `return` statement do? I think you need to re-raise the original failure here, not pretend there was success.

Reply:
Okay, Got it